graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java

```diff
@@ -103,35 +100,21 @@
     @NotNull
     public abstract ZonedDateTime creationDate();
 
-    // TODO 2.2: Migrate config setting to index set!
     @JsonProperty("index_analyzer")
-    @JsonIgnore
-    public String indexAnalyzer() {
-        return "standard";
-    }
+    @NotBlank
+    public abstract String indexAnalyzer();
 
-    // TODO 2.2: Migrate config setting to index set!
     @JsonProperty("index_template_name")
-    @JsonIgnore
-    public String indexTemplateName() {
-        return "graylog-internal-" + requireNonNull(id());
```

Comment:
This and all other settings need a migration to not break existing index sets in a 2.2 setup.

We also need UI components and/or good defaults, otherwise the creation of new index sets will not work.